package/kernel/linux/modules/netdevices.mk

```diff
@@ -418,6 +418,24 @@ endef
 $(eval $(call KernelPackage,switch-ip17xx))
 
 
+define KernelPackage/switch-qca8k
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=QCA8k device support
+  KCONFIG:= \
+	CONFIG_NET_DSA_QCA8K \
+	CONFIG_NET_DSA_TAG_QCA=y \
+	CONFIG_REGMAP=y
+  FILES:=$(LINUX_DIR)/drivers/net/dsa/qca8k.ko
```

Comment:
This does not compile on many targets. Many fail like this in the github CI:
````
ackaged contents of /__w/openwrt/openwrt/openwrt/build_dir/target-arm-openwrt-linux-muslgnueabi_musl/linux-imx_cortexa7/packages/ipkg-arm_cortex-a7_neon-vfpv4/kmod-switch-ip17xx into /__w/openwrt/openwrt/openwrt/bin/targets/imx/cortexa7/packages/kmod-switch-ip17xx_5.15.98-1_arm_cortex-a7_neon-vfpv4.ipk
ERROR: module '/__w/openwrt/openwrt/openwrt/build_dir/target-arm-openwrt-linux-muslgnueabi_musl/linux-imx_cortexa7/linux-5.15.98/drivers/net/dsa/qca8k.ko' is missing.
make[2]: *** [modules/netdevices.mk:436: /__w/openwrt/openwrt/openwrt/bin/targets/imx/cortexa7/packages/kmod-switch-qca8k_5.15.98-1_arm_cortex-a7_neon-vfpv4.ipk] Error 1
time: package/kernel/linux/compile#32.84#12.77#41.19
Error: Process completed with exit code 2.
````

This is probably happening on the targets which do not activate DSA. There package should not get activated. I thought OpenWrt will detect that automatically.

Reply:
Actually on 5.15 the driver is moved to qca directory... that should be handled
`$(LINUX_DIR)/drivers/net/dsa/qca/qca8k.ko`